packages/server-wallet/src/objectives/objective-manager.ts

```diff
@@ -54,87 +66,58 @@ export class ObjectiveManager {
 
     while (attemptAnotherProtocolStep) {
       await this.store.lockApp(channelToLock, async tx => {
-        let executeProtocol: () => ChannelState.ProtocolResult;
-        let protocolState: OpenChannelProtocol.ProtocolState | CloseChannelProtocol.ProtocolState;
+        let nextAction: ChannelState.ProtocolResult;
+        let protocolState: SupportedProtocolState;
 
         if (objective.type === 'OpenChannel') {
-          protocolState = await OpenChannelProtocol.getOpenChannelProtocolState(
+          protocolState = await getOpenChannelProtocolState(
             this.store,
             objective.data.targetChannelId,
             tx
           );
-          executeProtocol = () =>
-            OpenChannelProtocol.protocol(protocolState as OpenChannelProtocol.ProtocolState);
+          nextAction = recordFunctionMetrics(
+            openChannelProtocol(protocolState as OpenChannelProtocolState),
+            this.timingMetrics
+          );
         } else if (objective.type === 'CloseChannel') {
-          protocolState = await CloseChannelProtocol.getCloseChannelProtocolState(
+          protocolState = await getCloseChannelProtocolState(
             this.store,
             objective.data.targetChannelId,
             tx
           );
-          executeProtocol = () =>
-            CloseChannelProtocol.protocol(protocolState as CloseChannelProtocol.ProtocolState);
+          nextAction = recordFunctionMetrics(
+            closeChannelProtocol(protocolState as OpenChannelProtocolState),
```

Comment:
Same comment here about casting, although I am surprised that this isn't an error?